### ISSUE TITLE
7TV emotes 403 Forbidden

### ISSUE THREAD

**Issue by asedim** [lay295/TwitchDownloader#286]
Downloading chat with `--embed-emotes` or rendering it results in 403 Forbidden error.

```
$ ./TwitchDownloaderCLI --mode ChatDownload --id x --embed-emotes true --output chat.json
[STATUS] - Downloading 98%
[STATUS] - Downloading + Embedding Emotes
Unhandled exception. System.AggregateException: One or more errors occurred. (The remote server returned an error: (403) Forbidden.)
 ---> System.Net.WebException: The remote server returned an error: (403) Forbidden.
   at System.Net.HttpWebRequest.GetResponse()
   at System.Net.WebClient.GetWebResponse(WebRequest request)
   at System.Net.WebClient.DownloadBits(WebRequest request, Stream writeStream)
   at System.Net.WebClient.DownloadDataInternal(Uri address, WebRequest& request)
   at System.Net.WebClient.DownloadString(Uri address)
   at System.Net.WebClient.DownloadString(String address)
   at TwitchDownloaderCore.TwitchHelper.GetThirdPartyEmotes(Int32 streamerId, String cacheFolder, Emotes embededEmotes, Boolean bttv, Boolean ffz, Boolean stv)
   at TwitchDownloaderCore.ChatDownloader.<>c__DisplayClass3_1.<DownloadAsync>b__1()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location where exception was thrown ---
   at TwitchDownloaderCore.ChatDownloader.DownloadAsync(IProgress`1 progress, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at TwitchDownloaderCLI.Program.DownloadChat(Options inputOptions)
   at TwitchDownloaderCLI.Program.Main(String[] args)
Aborted (core dumped)
```

It seems that it's caused by the 7TV API
```
$ curl -s -o /dev/null -w "%{http_code}" "https://api.7tv.app/v2/emotes/global"
403
```

When visiting from the browser I get the CloudFlare's captcha
![image](https://user-images.githubusercontent.com/11590250/153769947-04d7ca30-9e95-46c2-bc3a-1d1d38b3538d.png)

I've been using TwitchDownloader for a long time and it's the first time it happened to me and I keep running into this every time now.

I see that the request exceptions from the 7TV API are not handled
https://github.com/lay295/TwitchDownloader/blob/1cc77742a926d8c3429f04b08c053983f60ff642/TwitchDownloaderCore/TwitchHelper.cs#L260
Would that be possible for you to handle them? It would be at a cost of missing some emotes but the whole process would have worked in a case like this.

**Comment by lay295** [comment]
Is it still happening? I would try and figure out why you're hitting the cloudflare captcha tbh. Maybe it was just the temporary "under attack mode" that 7tv enabled because they were getting DDOSd or maybe for some reason your IP is flagged by cloudflare?

I could just in code ignore the failure to get the 7TV emotes but I don't know if that's really desirable behavior, maybe I could just throw a warning.

**Comment by asedim** [comment]
I found the reason why I'm getting dinged by Cloudflare. I'm on a shared IP and it changed the day I started having those issues. I've reached out to my ISP to resolve this issue but in the meantime I can't render any chat. I'm thinking of getting ahold of some VPN or use mobile data.

It was mostly my problem and I know now how to work around it but it puts a perspective on what's desired in this situation. For me I think I would have wanted a warning and a passing flow. Maybe with a `--ignore-optional-emotes-errors` flag to have the best of two worlds but that might be too much to ask. I'll leave it up to you if you want to close the issue, consider a minor inconvenience and leave for later or handle the exception. In any case thanks for taking your time to reply.
